lib/ui/instrumentpanel.js

```diff
@@ -53,11 +53,32 @@ InstrumentPanel.prototype.setPage = function(page) {
   }
   if (page === this.pages.length && page < 10) {
     this.pages[page] = newPage();
+    this.populateNewPage(page);
   }
   this.currentPage = page;
   this.pushGridChanges();
 }
 
+InstrumentPanel.prototype.populateNewPage = function(page) {
+  var that = this;
+  const newPage = page;
+  this.pages.map(function(page) { // scan all pages to find existing widget
+    page.widgets.map(function(existingWidget) {
+      const foundKey = that.pages[newPage].widgets.find(function(widgetOnNewPage) {
+        return existingWidget.options.key === widgetOnNewPage.options.key;
+      });
+      if (! foundKey) {
+        var module = widgetModuleFor(existingWidget.options.path) || widgetModules[0];
+        var newOptions = Object.assign({}, existingWidget.options);
```

Comment:
consider object spread instead of Object.assign: `= {...existingWidget.options)`